common.c

```diff
@@ -944,10 +945,59 @@ static bool handleClientData(X11ConnData* data)
 			log_debug2(" XChangeProperty: property=%"PRIuCARD32" type=%"PRIuCARD32" format=%d)\n", req->property, req->type, req->format);
 			if (req->type == XA_WM_SIZE_HINTS)
 			{
+				// all properties in data->buf are req->format (32) bits long, 
+				// but flag property in XSizeHints is a long which is 64 bit long on 64 bit machines.
+				#ifdef LONG64
+				// offset the structure by 32 bits to align all of the other properties.
+				// this makes the flags property point into xChangePropertyReq on 64 bit
+				// so we have to get the flags separately.
+				XSizeHints* hints = (XSizeHints*)(data->buf + sz_xChangePropertyReq - 4);
+				#else
 				XSizeHints* hints = (XSizeHints*)(data->buf + sz_xChangePropertyReq);
+				#endif
```

Comment:
OK, the problem here is that we're using `XSizeHints` from `Xutil.h`, which is part of the Xlib API. We are not interacting with Xlib, so we should not be using any definitions from its headers... looks like we really just want to use `X.h` instead.

The data structure is described here:

https://tronche.com/gui/x/icccm/sec-4.html#s-4.1.2.3

I don't see a protocol-level declaration in the Xorg headers, so I guess hax11 needs to provide its own definition of the structure.